src/libs/vtkh/filters/PointTransform.cpp

```diff
@@ -5,61 +5,111 @@
 namespace vtkh
 {
 
+//---------------------------------------------------------------------------//
 PointTransform::PointTransform()
 {
   ResetTransform();
 }
 
+//---------------------------------------------------------------------------//
 PointTransform::~PointTransform()
 {
 
 }
 
-void PointTransform::ResetTransform()
+//---------------------------------------------------------------------------//
+void
+PointTransform::ResetTransform()
 {
   vtkm::MatrixIdentity(m_transform);
 }
 
+//---------------------------------------------------------------------------//
 void
 PointTransform::SetTranslation(const double& tx,
                                const double& ty,
                                const double& tz)
 {
-  vtkm::Matrix<double,4,4> matrix  = vtkm::Transform3DTranslate(tx, ty, tz);
+  vtkm::Matrix<double,4,4> matrix = vtkm::Transform3DTranslate(tx, ty, tz);
   m_transform = vtkm::MatrixMultiply(m_transform, matrix);
 }
 
-void PointTransform::SetRotation(const double& angleDegrees,
-                                 const vtkm::Vec<double, 3>& axis)
+//---------------------------------------------------------------------------//
+void
+PointTransform::SetRotation(const double& angleDegrees,
+                            const double& axisX,
+                            const double& axisY,
+                            const double& axisZ)
 {
-  vtkm::Matrix<double,4,4> matrix = vtkm::Transform3DRotate(angleDegrees, axis);
+  vtkm::Matrix<double,4,4> matrix = vtkm::Transform3DRotate(angleDegrees,
+                                                            axisX,
+                                                            axisY,
+                                                            axisZ);
   m_transform = vtkm::MatrixMultiply(m_transform, matrix);
 }
 
-void PointTransform::SetTransform(const vtkm::Matrix<double, 4, 4>& mtx)
+//---------------------------------------------------------------------------//
+void
+PointTransform::SetTransform(const double *matrix_values)
+{
+  // TODO: row vs col vs matrix vs array, what is the best
+  //       order for users to provide flat values to matrix?
```

Comment:
Note: this order seems good.

I googled some example transform matrices and threw them in easily.

Reply:
scale then rotate then translate

_T * R * S_

I didn't look at the code but is that what this is about? https://gamedev.stackexchange.com/questions/16719/what-is-the-correct-order-to-multiply-scale-rotation-and-translation-matrices-f